### PR TITLE
Expose debug only on localhost or authenticated

### PR DESCRIPTION
**Pull request by howardjohn**
Spinning off https://github.com/istio/istio/pull/31609 to get some
feedback. Basically, instead of just exposing on localhost, also allow
it when authenticated.

This is basically the same as the XDS port, but allows accessing it over
HTTP. Hopefully this makes some use cases a bit simpler.

### CODE REVIEW

**Review by irisdingbj — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Review by irisdingbj — APPROVED** [review]

File: pilot/pkg/xds/debug.go [hunk 1]
```diff
@@ -139,69 +140,116 @@ func (s *DiscoveryServer) InitDebug(mux *http.ServeMux, sctl *aggregate.Controll
...
+	s.addDebugHandler(mux, internalMux, "/debug/list", "List all supported debug commands in json", s.List)
 }
 
-func (s *DiscoveryServer) addDebugHandler(mux *http.ServeMux, path string, help string,
-	handler func(http.ResponseWriter, *http.Request)) {
+func (s *DiscoveryServer) addDebugHandler(mux *http.ServeMux, internalMux *http.ServeMux,
+	path string, help string, handler func(http.ResponseWriter, *http.Request)) {
 	s.debugHandlers[path] = help
-	mux.HandleFunc(path, handler)
+	// Add handler without auth. This mux is never exposed on an HTTP server and only used internally
```
Comment (ramaraochavali) [inline pilot/pkg/xds/debug.go:206]: Who uses these handlers internally?

Comment (ramaraochavali) [reply pilot/pkg/xds/debug.go:206]: nm.. Looks like DebugGen uses it.

File: pilot/pkg/xds/debug.go [hunk 2]
```diff
@@ -139,69 +140,116 @@ func (s *DiscoveryServer) InitDebug(mux *http.ServeMux, sctl *aggregate.Controll
 		ServiceDiscovery: s.MemRegistry,
 		Controller:       s.MemRegistry.Controller,
 	})
-	s.AddDebugHandlers(mux, enableProfiling, fetchWebhook)
+	internalMux := http.NewServeMux()
```
Comment (ramaraochavali) [inline pilot/pkg/xds/debug.go:143]: We could have stored in internalMux in Server struct so that we do not have to pass it around

**Comment by ramaraochavali** [comment]
What is the impact of this change on this ENABLE_DEBUG_ON_HTTP? With that flag on debug interfaces are enabled outside of localhost also without auth - I think this change does not allow that which is fine. I do not see a value of this flag now..

**Comment by howardjohn** [comment]
Yeah we can probably remove ENABLE_DEBUG_ON_HTTP in a followup, if we want, since it should be secure now in all(?) cases.

One issue may be if you have some sidecar in front of istiod so requests come from "localhost"

**Review by ericvn — APPROVED** [review]

**Comment by ramaraochavali** [comment]
> One issue may be if you have some sidecar in front of istiod so requests come from "localhost"
Yes.

Let us do a follow-up to cleanup that part

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**